tests/integration/records/test_records_api.py

```diff
@@ -72,3 +76,52 @@ def test_download_local_file(isolated_app):
 
         assert 1 == len(documents)
         assert 1 == len(files)
+
+
+def test_create_does_not_save_zombie_identifiers_if_record_creation_fails(isolated_app):
+    invalid_record = {
+        '$schema': 'http://localhost:5000/schemas/records/hep.json',
+        '_collections': [
+            'Literature',
+        ],
+        'control_number': 1936477,
+    }
+
+    with pytest.raises(ValidationError):
+        InspireRecord.create(invalid_record)
+
+    record_identifier = RecordIdentifier.query.filter_by(recid=1936477).one_or_none()
+    persistent_identifier = PersistentIdentifier.query.filter_by(pid_value='1936477').one_or_none()
+
+    assert not record_identifier
+    assert not persistent_identifier
+
+
+def test_create_does_not_save_zombie_records_if_record_minting_fails(isolated_app):
```

Comment:
actually we don't need this case because it's already covered by calling ``mint`` before ``.create``. If creation fails in ``PersistentIdentifier`` or ``RecordIdentifier`` will never create the record. WDYT?

Reply:
I was thinking of it more like a regression test just in case someone touches that part of the code again. What do you think?

Reply:
hmm it's bit arbitrary, when you have a ``pid``  you should have a ``RecordIdentifier``. In this case, we assume that we don't have one which is a wrong assumption. IMO this test is not necessary since we are not testing any real case.

Reply:
ok!